cmake/kokkos_arch.cmake

```diff
@@ -663,9 +663,19 @@ IF (KOKKOS_ENABLE_OPENMPTARGET)
 ENDIF()
 
 IF (KOKKOS_ENABLE_OPENACC)
-  COMPILER_SPECIFIC_FLAGS(
-    NVHPC -acc
-  )
+  IF(KOKKOS_CUDA_ARCH_FLAG)
+    STRING(REPLACE "sm_" "cc" NVHPC_CUDA_ARCH ${KOKKOS_CUDA_ARCH_FLAG})
+    COMPILER_SPECIFIC_FLAGS(
+      NVHPC -acc -gpu=${NVHPC_CUDA_ARCH}
+    )
+  ELSEIF(KOKKOS_AMDGPU_ARCH_FLAG)
+    # FIXME_OPENACC
+    COMPILER_SPECIFIC_FLAGS(
+      NVHPC -acc
+    )
+  ELSE()
+    MESSAGE(SEND_ERROR "Must specify a GPU architecture with OpenACC")
```

Comment:
OpenACC does not requires target device type/architecture at compile time, which can be set at runtime using an environment variable (`ACC_DEVICE_TYPE`) or the implementation can choose a default device type.
For NVHPC compiler, `-gpu` option is optional; if not set, the compiler will generate binary for all available (and supported) architectures.
Therefore, an alternative option for `ELSE()` case is to let the NVHPC compiler to choose available device architecture (`NVHPC -acc`).

Reply:
I am fine withdrawing these changes for now.
With the changes in this first version I needed to know at compile-time whether we were targeting NVIDIA or AMD gpu but I will go for `acc_default_device` and drop this for now.
Let's bring it up with the team at the next developer meeting.

Reply:
What I suggest is not to remove all the architecture-specific options (`-gpu`); instead make it optional.
Then, the code will look like this:

IF(KOKKOS_CUDA_ARCH_FLAG)
     STRING(REPLACE "sm_" "cc" NVHPC_CUDA_ARCH ${KOKKOS_CUDA_ARCH_FLAG})
     COMPILER_SPECIFIC_FLAGS(
       NVHPC -acc -gpu=${NVHPC_CUDA_ARCH}
     )
ELSE()
     COMPILER_SPECIFIC_FLAGS(
       NVHPC -acc
     )
ENDIF()

Reply:
Got it.  Will update.